lib/common/components/forms/custom_text_field.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_boilerplate/common/components/forms/custom_date_picker.dart';
+import 'package:flutter_boilerplate/common/components/forms/custom_form_input_class.dart';
+
+class CustomTextField extends StatefulWidget {
+  final CustomFormInput input;
+
+  const CustomTextField({
+    Key? key,
+    required this.input,
+  }) : super(key: key);
+
+  @override
+  State<CustomTextField> createState() => _CustomTextFieldState();
+}
+
+class _CustomTextFieldState extends State<CustomTextField> {
+  bool _error = false;
+
+  void validateField() {
+    setState(() {
+      _error =
+          widget.input.controller.text == 'a' && widget.input.errorMessage != ''
```

Comment:
what's a?

Reply:
that's for testing the error message, it was supposed to check the regex there

Reply:
fixed it, now it checks the regex there